File: ryx-python/ryx/cli/style.py
"""
Ryx CLI — color and formatting utilities.

Usage::

    from ryx.cli.style import PREFIX, OK, FAIL, WARN, green, cyan

    print(f"{PREFIX} {OK} Migration applied")
    print(f"{PREFIX} {FAIL} {red('Error:')} something broke")
"""

from __future__ import annotations

import os
import sys

_RESET = "\033[0m"
_BOLD = "\033[1m"
_RED = "\033[31m"
_GREEN = "\033[32m"
_YELLOW = "\033[33m"
_BLUE = "\033[34m"
_MAGENTA = "\033[35m"
_CYAN = "\033[36m"
_GREY = "\033[90m"


def _supports_color() -> bool:
    if not sys.stdout.isatty():
        return False
    if os.environ.get("NO_COLOR"):
        return False
    term = os.environ.get("TERM", "")
    if term == "dumb":
        return False
    return True


_USE_COLOR = _supports_color()


def _c(text: str, code: str) -> str:
    return f"{code}{text}{_RESET}" if _USE_COLOR else text


def bold(text: str) -> str:
    return _c(text, _BOLD)


def dim(text: str) -> str:
    return _c(text, _GREY)


def red(text: str) -> str:
    return _c(text, _RED)


def green(text: str) -> str:
    return _c(text, _GREEN)


def yellow(text: str) -> str:
    return _c(text, _YELLOW)


def blue(text: str) -> str:
    return _c(text, _BLUE)


def magenta(text: str) -> str:
    return _c(text, _MAGENTA)


def cyan(text: str) -> str:
    return _c(text, _CYAN)


PREFIX = _c("[ryx]", f"{_BLUE}{_BOLD}") if _USE_COLOR else "[ryx]"
OK = _c("✓", _GREEN) if _USE_COLOR else "✓"
FAIL = _c("✗", _RED) if _USE_COLOR else "✗"
WARN = _c("⚠", _YELLOW) if _USE_COLOR else "⚠"

File: ryx-rs/src/cli/style.rs
//! ANSI color helpers for the Ryx CLI.
//!
//! Automatically disables colours when ``NO_COLOR`` is set, ``TERM=dumb``,
//! or stdout is not a TTY.

use std::io::IsTerminal;

const BOLD: &str = "\x1b[1m";
const RESET: &str = "\x1b[0m";
const RED: &str = "\x1b[31m";
const GREEN: &str = "\x1b[32m";
const YELLOW: &str = "\x1b[33m";
const BLUE: &str = "\x1b[34m";
const MAGENTA: &str = "\x1b[35m";
const CYAN: &str = "\x1b[36m";

fn use_colour() -> bool {
    if std::env::var("NO_COLOR").is_ok() {
        return false;
    }
    if matches!(std::env::var("TERM").as_deref(), Ok("dumb")) {
        return false;
    }
    std::io::stdout().is_terminal()
}

macro_rules! colour {
    ($name:ident, $code:expr) => {
        pub fn $name(s: &str) -> String {
            if use_colour() {
                format!("{}{}{}", $code, s, RESET)
            } else {
                s.to_string()
            }
        }
    };
}

colour!(red, RED);
colour!(green, GREEN);
colour!(yellow, YELLOW);
colour!(cyan, CYAN);
colour!(magenta, MAGENTA);

pub fn prefix() -> String {
    if use_colour() {
        format!("{BOLD}{BLUE}[ryx]{RESET}")
    } else {
        "[ryx]".to_string()
    }
}

pub fn ok_mark() -> String {
    if use_colour() {
        format!("{GREEN}\u{2713}{RESET}")
    } else {
        "\u{2713}".to_string()
    }
}

pub fn fail_mark() -> String {
    if use_colour() {
        format!("{RED}\u{2717}{RESET}")
    } else {
        "\u{2717}".to_string()
    }
}

pub fn warn_mark() -> String {
    if use_colour() {
        format!("{YELLOW}\u{26A0}{RESET}")
    } else {
        "\u{26A0}".to_string()
    }
}
